vignettes/adrs_gcig.Rmd

```diff
@@ -0,0 +1,799 @@
+  ~"Confirmed RECIST1.1 response at 1st time point",
+  "CR", "CR", "CR",
+  "CR", "PR/SD/PD", "PD",
+  "CR", "NE", "NE",
+  "PR", "CR/PR", "PR",
+  "PR", "SD/PD/NE", "SD/PD/NE",
+  "CR/PR", "missing", "SD",
+  "SD/PD/NE", "any", "SD/PD/NE"
+)
+
+knitr::kable(list_resp)
+```
+
+In brief, the following steps are described below:
+
+1. Derive confirmed overall response based on the response from next consecutive assessment.
+2. Display a warning if responses used for confirmation are less than confirmed_period apart.
+3. Remove unnecessary variables, assign `AVAL`, `AVALC` and other parameter details.
+
+
+```{r}
+confirmation_period <- 21
+
+covr_r11 <- ovr_r11 %>%
+    arrange(USUBJID, ADTM) %>%
+    group_by(USUBJID) %>%
+    mutate(
+      AVALC.next = lead(AVALC),
+      AVAL.next = lead(AVAL),
+      ADT.next = lead(ADT)
+    ) %>%
+    ungroup() %>%
+    mutate(AVALC.confirmed = case_when(
+      # CR
+      AVALC == "CR" & AVALC.next == "CR" ~ "CR",
+      AVALC == "CR" & AVALC.next %in% c("PR", "SD", "PD") ~ "PD",
+      AVALC == "CR" & AVALC.next == "NE" ~ "NE",
+      # PR
+      AVALC == "PR" & AVALC.next %in% c("CR", "PR") ~ "PR",
+      AVALC == "PR" & AVALC.next %in% c("SD", "PD", "NE") ~ AVALC.next,
+      # next assessment NA
+      AVALC %in% c("CR", "PR") & is.na(AVALC.next) ~ "SD",
+      # no need to confirm SD, PD, NE
+      AVALC %in% c("SD", "PD", "NE") ~ AVALC,
+      !is.na(AVALC) ~ AVALC
+    ))
+
+covr_r11_check <- covr_r11 %>%
+  mutate(diff_days = as.numeric(difftime(ADT.next, ADT, units = "days"))) %>%
+  filter(diff_days < confirmation_period) %>%
+  mutate(warn = paste(
+    "For USUBJID", USUBJID, "to confirm", AVISIT,
+    "visit, a visit that took place", diff_days, "days later was used."
+)) %>% pull(warn)
+
+if (length(covr_r11_check) > 0) {
+  cli::cli_warn("{covr_r11_check}")
+}
+
+covr_r11 <- covr_r11 %>%
+  select(-AVAL, -AVALC, -AVAL.next, -AVALC.next, -ADT.next) %>%
+  rename(AVALC = AVALC.confirmed) %>%
+  mutate(AVAL = aval_resp(AVALC)) %>%
+  mutate(
+    PARAMCD = "COVRR11",
+    PARAM = "Confirmed RECIST 1.1 Overall Response by Investigator",
+    PARAMN = 7,
+    PARCAT1 = "Investigator",
+    PARCAT1N = 1,
+    PARCAT2 = "RECIST 1.1",
+    PARCAT2N = 2,
+)
+  
+adrs <- bind_rows(adrs, covr_r11)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, AVALC, ADT),
+  filter = PARAMCD == "COVRR11" & AVALC != "MISSING"
+)
+```
+
+## Combined Best Confirmed Overall Response {#combcbor}
+
+Now that we have both components of the combined response as confirmed responses, we can derive the combined response and at a later step Combined Best Confirmed Overall Response parameter.
+
+Following table provides a summary of the `Combined Overall Response by Investigator (Confirmation of RECIST1.1 Reqired)` calculation at each time point as per GCIG guidelines.
+
+```{r echo=FALSE}
+list_resp <- tribble(
```

Comment:
I will not suggest to put this table in the vignette as it will be very confusing as doesn't matches with GCIG. GCIG also didn't give for overall confirmed response..its based on taget, no target , new lesion at each visit.

Reply:
GCIG Table 3 can be redefined using overall response, and it seems to me that we should do so and it is not risky. The set target/non-target/new clearly determines the overall response.
The other thing is that Table 3 does not cover all cases, and this is where the problem arises.